server/src/server.rs

```diff
@@ -765,52 +767,32 @@ impl CubebServer {
         });
 
         let latency = params.latency_frames;
+
+        let server_stream = &mut self.streams[stm_tok];
         assert!(size_of::<Box<ServerStreamCallbacks>>() == size_of::<usize>());
-        let user_ptr = cbs.as_ref() as *const ServerStreamCallbacks as *mut c_void;
-
-        unsafe {
-            context
-                .stream_init(
-                    stream_name,
-                    input_device,
-                    input_stream_params,
-                    output_device,
-                    output_stream_params,
-                    latency,
-                    Some(data_cb_c),
-                    Some(state_cb_c),
-                    user_ptr,
-                )
-                .and_then(|stream| {
-                    if self.streams.len() == self.streams.capacity() {
-                        trace!(
-                            "server connection ran out of stream slots. reserving {} more.",
-                            STREAM_CONN_CHUNK_SIZE
-                        );
-                        self.streams.reserve_exact(STREAM_CONN_CHUNK_SIZE);
-                    }
+        let user_ptr = server_stream.cbs.as_ref() as *const ServerStreamCallbacks as *mut c_void;
+
+        let stream = unsafe {
+            let stream = context.stream_init(
```

Comment:
In pulse-backend, will `stream_init` return `CUBEB_ERROR`, or any non `CUBEB_OK` when `PulseStream::new::check_error` is fired with a `CUBEB_STATE_ERROR`? Just want to be sure we will return `return Err(e.into())` in this case.

Reply:
I believe it will, if it happens during stream initialization.  `check_error` remains registered for the lifetime of the stream, so if it runs after stream init, it'll just run the state callback and leave it up to the caller to handle the stream error.